cai_causal_graph/time_series_causal_graph.py

```diff
@@ -681,7 +673,43 @@ def add_edge(
                 )
         # No else needed as we don't need to check time direction for other edge types.
 
-        edge = super().add_edge(source, destination, edge_type=edge_type, meta=meta, edge=edge, **kwargs)
+    @_reset_ts_graph_attributes
+    def add_edge(
+        self,
+        /,
+        source: Optional[NodeLike] = None,
+        destination: Optional[NodeLike] = None,
+        *,
+        edge_type: EdgeType = EdgeType.DIRECTED_EDGE,
+        meta: Optional[dict] = None,
+        edge: Optional[Edge] = None,
+        **kwargs,
+    ) -> Edge:
+        """
+        Add an edge to the graph. See `cai_causal_graph.causal_graph.CausalGraph.add_edge` for more details.
+
+        In addition to the `cai_causal_graph.causal_graph.CausalGraph.add_edge` method, this method also populates the
+        metadata of the nodes with the variable name and the time lag.
+
+        :param source: The source node of the edge.
+        :param destination: The destination node of the edge.
+        :param edge_type: The type of edge to add.
+        :param meta: The metadata to add to the edge.
+        :param edge: The edge to add.
+        :param kwargs: Additional keyword arguments to pass to the
+            `cai_causal_graph.causal_graph.TimeSeriesCausalGraph.add_edge` method.
+        :return: The edge that was added.
+        """
+        # if the nodes are Node but not TimeSeriesNodes, covert them to TimeSeriesNodes
+        if edge is not None and isinstance(edge.source, Node) and not isinstance(edge.source, TimeSeriesNode):
```

Comment:
`edge.destination` is not checked. Could be some weird scenario where `source` is `Node` and `destination` is `TimeSeriesNode`. Would recommend the following:
```
if edge is not None:
   assert isinstance(edge, _EdgeCls), 'NICE ERROR MESSAGE'
   # Check source node type
   source_node = edge.source
   if isinstance(source_node, Node) and not isinstance(source_node, _NodeCls):
      source_node = _NodeCls.from_dict(source_node.to_dict())
   # Check destination node type
   destination_node = edge.destination
   if isinstance(destination_node, Node) and not isinstance(destination_node, _NodeCls):
      destination_node = _NodeCls.from_dict(destination_node.to_dict())
   # See if either node was recreated, if yes, then create new edge with appropriate nodes
   if source_node != edge.source or destination_node != edge.destination:
      edge = _EdgeCls(source_node, destination_node, edge_type=edge.edge_type, edge.meta)
```